### ISSUE TITLE
Remove extension from migration names

### ISSUE THREAD

**Issue by glensc** [knex/knex#4688]
So, having your migrations in TypeScript, running them via `ts-node` locally works fine, but when deploying production the migrations are compiled to `.js` files via `tsc`.

So migrations recorded in the production database have `.js` extension, and when such database dump is attempted to be used in the development, you'll have name conflict as seems like migrations not applied.

Workaround is this:
```sql
update knex_migrations set name=replace(name, '.js', '.ts');
```

But I was thinking, as the project is heading towards 1.0, perhaps make the breaking change and rename the extension from migration names?

**Comment by elhigu** [comment]
I agree that not having migration file extension in the database would be a good thing.

**Comment by glensc** [comment]
implementation wise, the code could fetch name from database, strip extension, then compare. i.e not do `select * from knex_migrations where name='123_111.js'` but rather do `select` without `where`.

**Comment by kibertoad** [comment]
@glensc Would you consider sending a PR for this?

**Comment by elhigu** [comment]
I thinkg that implementation sounds also good. It should be backwards compatible. I'm not sure about the case where the same directory has migration_name.ts and migration_name.js available and both extensions enabled in configuration. 

Should that throw an error or just silently select either one of those to run. In any case it is probably good to have complete file name in the database and just ignore extension in comparison 👍 That will make distinction in those cases which file actually was applied to DB if there are many with the same base name.

**Comment by mercmobily** [comment]
Closing this in favour of #2756 which covers the same issue
